bluesky/plan_stubs.py

```diff
@@ -1020,3 +1020,50 @@ def repeated_plan():
                     yield Msg('sleep', None, d)
 
     return (yield from repeated_plan())
+
+
+def update_epicsmotor_user_offset(epicsmotor, val):
+    '''Follows the correct procedure to update an EpicsMotor user offset.
+
+    There is a complex set of steps involved in correctly updating the user
+    offset value as the effect of changing its value depends on he values of
+    other signals, this plan_stub genreates the instructions to perform those
+    steps. See the section on Calibration-related fileds at https://epics.anl.gov/bcda/synApps/motor/R4-1/motorRecord.html>>>>>>>ccb1ced23ad740f7c7c0e9a5a312643f125c1599
+    for a detailed description of the how this is meant to work.
+
+    Parameters
+    ----------
+    epicsmotor : ``ophyd.EpicsMotor`` instance
+        This is an EpicsMotor instance whose user-offset is to be changed
+    val : float
+        The new value for the user offset.
+    '''
+    if hasattr(epicsmotor, 'user_offset') and hasattr(epicsmotor,
```

Comment:
I would not put these behind a conditional, if you pass in a non-epics motor to this (or the other one) they _should_ blow up as they are not going to do what you think they will do.  

For example if you pass a mover in, expecting it to reset the origin to where ever you currently are (say 1000) and then tell it to move to 1 expecting it to move one unit forward but instead it moves 999 units the other way.